cpp/src/arrow/engine/substrait/options.h

```diff
@@ -67,6 +68,10 @@ using NamedTableProvider =
     std::function<Result<compute::Declaration>(const std::vector<std::string>&)>;
 static NamedTableProvider kDefaultNamedTableProvider;
 
+using NamedTapKindMapper = std::function<Result<std::string>(const std::string&)>;
+static NamedTapKindMapper kDefaultNamedTapKindMapper =
```

Comment:
If you wanted to add this, I think we should probably match the signature of kDefaultNamedTableProvider and returns a Declaration instead?

Reply:
A string-typed argument is not sufficient for creating a declaration for an internal (non-source) node - the input declarations are required, and probably options too. Another reason for the differing signatures is that here the string-typed parameter is for the tap kind, which is semantically different than a table name.